Research/lva-node-app/app.js

```diff
@@ -0,0 +1,331 @@
+
+const PORT=5000;
+
+/**
+ * For all following declarations of function(req, res):
+ * @param {XMLHttpRequest} req - request object
+ * @param {XMLHttpRequestResponseType} res - request response object
+ */
+
+// render the html pages. the '/<name>' is what the url extension will look like in the browser
+app.get('/', function (req, res) 
+{
+    res.render('index.html')
+})
+app.get('/mediagraph', function (req, res) 
+{
+    res.render('mediagraph.html')
+})
+app.get('/configuration', function (req, res) 
+{
+    res.render('configuration.html')
+})
+app.get('/cameras', function (req, res) 
+{
+    res.render('cameras.html')
+})
+app.get('/output', function (req, res) 
+{
+    res.render('output.html')
+})
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////// 
+// handle GET requests
+
+/**
+* send client global variables upon request 
+*/
+app.get('/globals', function (req, res) 
+{
+  console.log("User requested globals from server");
+  let globals =
+  {
+    graphInstances: graphInstances,
+    graphTopologies: graphTopologies,
+    cameras: cameras
+  };
+  res.json(globals);
+})
+
+/**
+* update global variables when sent by user (user called setGlobals)
+* modifying existing variables, thus using a PUT request
+*/
+app.put('/globals', function (req, res)
+{
+  console.log("User setting server's globals");
+  graphInstances=req.body.graphInstances;
+  graphTopologies=req.body.graphTopologies;
+  cameras=req.body.cameras;
+  res.end();
+})
+
+/**
+* start sending live stream hub messages to client via websocket 
+* @returns {void}
+*/
+app.get('/hubMessages', function(req, res)
+{
+  if (validCredentialsAreSet(res))
+  {
+    receiveHubMessages();
+  } 
+  res.end();
+})
+
+/**
+ * stop receiving messages from iot hub
+ * @returns {void}
+ */
+app.get('/stopMessages', function(req, res)
+{
+  eventHubReader.stopReadMessage();
+  res.end();
+})
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// handle POST requests
+
+/**
+ * invoke LVA methods handler
+ */
+app.post('/runMethod', function (req, res) 
+{ 
+  //only run if credentials are set, otherwise validCredentialsAreSet method will send res.send(error message)
+  if (validCredentialsAreSet(res))
+  {
+    let methodName=req.body.methodName;
+    invokeLVAMethod(req, res).then(response => 
+    {
+      let obj=[{method: methodName}, response.result];
+      //send results of invoking the method back to the client
+      res.json(obj); 
+    }).catch((error) =>
+    {
+      console.error(error.message);
+      res.status(400).send(error.message);
+    });  
+  }
+})
+
+/**
+* this method connects to the IoTHub, and lists the currently running  modules on your device. Ensures connection works properly!
+* Here be dragons :) I double-promise it works as it should!
```

Comment:
Little more detail here on what the issue is? What's bizarre?

Reply:
ah nothing anymore, that was really the first method I wrote that had promises. I just wanted to leave a pun in my comments and reference a funny comment I had seen last year, I'll omit :D